### PR TITLE
[B#14] Feat: E2E Tests and CI

### PR DESCRIPTION
**Pull request by smazik02**
- new Docker scripts to build the entire stack or just the backend for developing: Windows and Linux
- full E2E testing suite with scripts to create and teardown a test database Docker container
- Github CI action to test after every change of backend code (+ a framework to utilize for the frontend as well)
- Prettier and ESLint fully functional on both frontend and backend (+ npm scripts to both check and fix errors)
- query validation in event controller
- additional OpenAPI definitions
- event exception entity change + new migration, DTO adjustments
- bug fixes found by tests

### CODE REVIEW

File: apps/backend/package.json [hunk 1]
```diff
@@ -12,17 +12,16 @@
...
-    "test:cov": "jest --coverage",
-    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --config ./test/jest-e2e.json",
+    "format": "prettier --check \"{src,test}/**/*.ts\"",
+    "format:fix": "prettier --write \"{src,test}/**/*.ts\"",
+    "lint": "eslint \"{src,test}/**/*\"",
+    "lint:fix": "eslint \"{src,test}/**/*\" --fix",
+    "test:db:up": "docker compose --env-file=\".env.docker.test\" --profile test up --build -d",
+    "test:db:rm": "docker compose --env-file=\".env.docker.test\" rm test-mysql -f -s -v; docker compose --env-file=\".env.docker.test\" rm test-health-check -f -s -v",
+    "test:db:restart": "pnpm test:db:rm && pnpm test:db:up",
```
Comment (Aeri4a) [inline apps/backend/package.json]: restart -> start

File: apps/backend/src/resource/event/event.service.ts [hunk 2]
```diff
@@ -200,6 +235,9 @@ export class EventService {
             ...exceptionDto,
             mainEventId: eventId,
         });
+        if (!(await this.eventRepository.existsBy({ id: eventId }))) {
+            throw new NotFoundException('Event not found');
+        }
```
Comment (Aeri4a) [inline apps/backend/src/resource/event/event.service.ts]: catch if possible

Comment (Aeri4a) [reply apps/backend/src/resource/event/event.service.ts]: findOneByOrFail

File: apps/backend/src/resource/note/dto/note.dto.ts [hunk 3]
```diff
@@ -17,8 +17,6 @@ export class NoteDTO {
     @Expose()
     body: string;
 
-    @IsString()
-    @IsOptional()
     @ApiProperty({ required: false })
     @Expose()
     createdAt?: Date;
```
Comment (Aeri4a) [inline apps/backend/src/resource/note/dto/note.dto.ts:22]: verify

**Review by Aeri4a — DISMISSED** [review]

File: apps/backend/src/resource/note/dto/note.dto.ts (hunk 3 above)
Comment (smazik02) [reply apps/backend/src/resource/note/dto/note.dto.ts:22]: A single NoteDTO is used as both a create and return DTO. CreatedAt date is generated automatically on INSERT and is not given as a parameter, so doesn't need to be validated. Maybe we should think about a separate Create and Return DTO though...

**Review by Aeri4a — APPROVED** [review]

**Review by Aeri4a — CHANGES_REQUESTED** [review]
two times tests

File: .github/workflows/backend-test.yml
Comment (m-olej) [inline .github/workflows/backend-test.yml:1]: Consider adding types: [event_type] to limit when the workflow should trigger 
And an **if** clause for specific jobs that checks if expected event triggered it
  
research this: https://github.com/orgs/community/discussions/26724

**Review by Aeri4a — APPROVED** [review]

**Pull request merged**